Biasing/src/Biasing/DeepEcalProcessFilter.cxx

```diff
@@ -0,0 +1,128 @@
+#include "Biasing/DeepEcalProcessFilter.h"
+
+/*~~~~~~~~~~~~*/
+/*   Geant4   */
+/*~~~~~~~~~~~~*/
+#include "G4EventManager.hh"
+#include "G4RunManager.hh"
+#include "G4Step.hh"
+#include "G4String.hh"
+#include "G4Track.hh"
+
+/*~~~~~~~~~~~~~*/
+/*   SimCore   */
+/*~~~~~~~~~~~~~*/
+#include "SimCore/UserEventInformation.h"
+#include "SimCore/UserTrackInformation.h"
+
+namespace biasing {
+
+bool hasProcessNeeded_{false};
+bool photonFromTarget_{false};
+bool hasDeepEcalProcess_{false};
+
+DeepEcalProcessFilter::DeepEcalProcessFilter(const std::string& name,
+                                     framework::config::Parameters& parameters)
+    : simcore::UserAction(name, parameters) {
+  bias_threshold_ = parameters.getParameter<double>("bias_threshold");
+  processes_ = parameters.getParameter<std::vector<std::string>>("processes");
+  ecal_min_Z_ = parameters.getParameter<double>("ecal_min_Z");
+  require_photon_fromTarget_ = parameters.getParameter<bool>("require_photon_fromTarget");
+}
+
+
+void DeepEcalProcessFilter::BeginOfEventAction(const G4Event* event) {
+  hasDeepEcalProcess_ = false;
+  photonFromTarget_ = false;
+}
+
+void DeepEcalProcessFilter::stepping(const G4Step* step) {
+  hasProcessNeeded_ = false;
+  
+  // Get the track associated with this step.
+  auto track{step->GetTrack()};
+  
+  // Check the creation process and PDG ID of the particle
+  auto processName = track->GetCreatorProcess() ? track->GetCreatorProcess()->GetProcessName() : "unknown";
+  auto PDGid = track->GetParticleDefinition()->GetPDGEncoding();
+  
+  // Skip the steps that are for the recoil electron
+  // PrimaryToEcalFilter made sure there is a fiducial e-
+  if (processName.contains("unknown")) return;
+  
+  // Energy of the particle is below threshold, move to next step
+  if (track->GetKineticEnergy() < bias_threshold_) {
+    return;
+  }
+  
+  // Check in which volume the particle is currently
+  auto volume{track->GetVolume()->GetLogicalVolume()
+    ? track->GetVolume()->GetLogicalVolume()->GetName()
+    : "undefined"};
+  
+  auto trackInfo{simcore::UserTrackInformation::get(track)};
+  // Tag the brem photon from the primary electron
+  if (processName.contains("eBrem") and (track->GetParentID() == 1)) {
+    trackInfo->tagBremCandidate();
+    getEventInfo()->incBremCandidateCount();
+    trackInfo->setSaveFlag(true);
+    if (volume.contains("target")) {
+      photonFromTarget_ = true;
+    }
+  }
+  
+  // If we require that the photon comes from the target and
+  // and if it does not, let's skip the event
+  if (require_photon_fromTarget_ and !photonFromTarget_) {
+    return;
+  }
+
+  // Tag if the event has the processes we are looking for
+  for (auto& process : processes_) {
+    // ldmx_log(debug) << "Allowed processed " << process << " now we have " << processName;
+    if (processName.contains(process)) {
+      hasProcessNeeded_ = true;
+      break;
+    }
+  }
+  
```

Comment:
```suggestion
  // skip this step if it does not have any of the processes needed
  if (not hasProcessNeeded) return;

```

The goal in moving this check earlier is two fold:
1. Keep a value's definition and usage close together so folks can see how it is used.
2. Avoid costly string comparisons below if they aren't needed.